### PR TITLE
feat(envfactory): shared BuildAcceptedSet helper for ReconcileCredentials

### PR DESCRIPTION
**Pull request by aaron-zeisler**
## Summary

- Adds `BuildAcceptedSet` in `internal/envfactory/reconcile_helper.go` — a pure, shared helper that both `autoconfig_actions.go` and `filedata_actions.go` will use to prepare inputs for `EnvContext.ReconcileCredentials`
- Closes [SDK-2546](https://launchdarkly.atlassian.net/browse/SDK-2546)

## Background

A follow-up will update both action handlers to call `BuildAcceptedSet(params)` and pass the result to `env.ReconcileCredentials(set)`. This PR delivers the helper itself, keeping it as a separate reviewable unit from the handler wiring.

The helper's contract:
- Builds `credential.AcceptedSet` from `params.AcceptedSDKKeys` / `params.AcceptedMobileKeys` (the arrays), keyed by credential value — not by the human-readable `key` identifier. A rename (same value, different identifier) is therefore a no-op.
- Adds and designates the anchor (`params.SDKKey`) via `WithPrimarySDKKey` and the primary mobile key (`params.MobileKey`) via `WithPrimaryMobileKey`. The builder de-duplicates by value, so an anchor/primary that also appears in its array is added once.
- If the anchor is undefined, no set is built and a `*credential.MalformedCredentialSetError` is returned with an empty `AcceptedSet`, signalling the caller to preserve the previous accepted state and (for RAC handlers) reconnect with jitter. There is no "anchor present in the array" check — add-and-designate guarantees the anchor is present, and structural validation of the wire payload happens upstream.
- Implements the "trust the array" expiry policy: the legacy `sdkKey.expiring` wire slot is never consulted; expiry comes only from `AcceptedSDKKey.Expiry` / `AcceptedMobileKey.Expiry`.
- Plumbs per-key mobile expiry through `WithExpiringMobileKey`, mirroring the SDK side, so an expiring mobile key lands as an expiring entry in the set.

## Changes

- **`internal/envfactory/reconcile_helper.go`** — `BuildAcceptedSet`, built on the `internal/credential` pointer builder (`NewAcceptedSetBuilder().…Build()`)
- **`internal/envfactory/reconcile_helper_test.go`** — unit tests covering the edge cases: happy path, multiple concurrent keys, rename no-op, de-expiry to permanent, anchor-absent-from-array (now accepted via add-and-designate), undefined-anchor error, no-SDK-key error, mixed add+re-anchor+remove, anchor-never-expiring invariant, multiple mobile keys, expiring mobile key, and trust-the-array policy

### Base branch

Stacked on `aaronz/SDK-2538/reconcile-credentials-api`; merge that first. Targets `feat/concurrent-keys` transitively.

[SDK-2546]: https://launchdarkly.atlassian.net/browse/SDK-2546?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

> [!NOTE]
> **Medium Risk**
> Touches credential acceptance and anchor/expiry rules that will feed reconciliation, but the change is a new pure helper with tests and is not yet wired into live handlers.
> 
> **Overview**
> Introduces **`BuildAcceptedSet`**, a shared `envfactory` helper that turns **`EnvironmentParams`** into a **`credential.AcceptedSet`** (plus anchor SDK key) for upcoming **`EnvContext.ReconcileCredentials`** use in autoconfig and filedata handlers. Handler wiring is intentionally deferred.
> 
> The builder treats credentials by **secret value** (renames are no-ops), designates anchor and primary mobile keys via **`WithPrimarySDKKey` / `WithPrimaryMobileKey`**, merges **`AcceptedSDKKeys` / `AcceptedMobileKeys`** with per-entry expiry, ignores the legacy **`sdkKey.expiring`** slot, and returns **`MalformedCredentialSetError`** when the anchor is missing so callers can keep prior state.
> 
> Adds broad unit tests for happy paths, concurrent keys, de-expiry, anchor invariants, mobile expiry, and malformed inputs.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 558b2e62d85c91aca82137fb69093558d727ed7c. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: internal/envfactory/reconcile_helper.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+// identifier). A rename — same value, different identifier — therefore produces the same
+// AcceptedSet as if nothing changed.
+//
+// Expiry comes from AcceptedSDKKey.Expiry / AcceptedMobileKey.Expiry (the arrays). The legacy
+// sdkKey.expiring wire slot is never consulted here ("trust the array", design §6.3).
+//
+// If the anchor is malformed, no set is built and a *relayenv.MalformedCredentialSetError is
+// returned with an empty AcceptedSet. The anchor is malformed when params.SDKKey is undefined
+// or is not found in AcceptedSDKKeys by value. The caller must preserve the previous accepted
+// state and, for RAC handlers, reconnect the stream with jitter to force a fresh put (design §9).
```
Comment (aaron-zeisler) [inline internal/envfactory/reconcile_helper.go]: Throughout the comments in this pull request, remove references to the design document sections and project-specific tasks.

Comment (aaron-zeisler) [reply internal/envfactory/reconcile_helper.go]: I would prefer if they were removed from the pull request's description, as well.

File: internal/envfactory/reconcile_helper.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	for _, k := range params.AcceptedMobileKeys {
+		if k.Expiry.IsZero() {
+			b.WithMobileKey(k.Value)
+		} else {
+			b.WithExpiringMobileKey(k.Value, k.Expiry)
+		}
+	}
+
+	// Designate the primary mobile key (the wire's mobKey). De-duplicated against the array above.
+	b.WithPrimaryMobileKey(params.MobileKey)
```
Comment (aaron-zeisler) [inline internal/envfactory/reconcile_helper.go]: Why not include this line just below line 37, in the chunk of code that's dealing with the other "primary" keys?

Also, is de-duplication (like lines 42-44) necessary in this function, or is the same de-duplication performed in the builder (I think it's already being done in the builder)?

**Review by keelerm84 — APPROVED** [review]
